test/zencode/foreach.bats

```diff
@@ -0,0 +1,118 @@
+load ../bats_setup
+load ../bats_zencode
+SUBDOC=foreach
+
+@test "Simplest loop ever" {
+    cat << EOF | save_asset verysimple.data
+{
+  "xs": ["a", "b"]
+}
+EOF
+
+    cat << EOF | zexe verysimple.zen verysimple.data
+Given I have a 'string array' named 'xs'
+When I create the new array
+Foreach 'x' in 'xs'
+When I insert 'x' in 'new array'
+EndForeach
+Then print 'new array' as 'string'
+EOF
+    save_output "very_simple.out"
+    assert_output '{"new_array":["a","b"]}'
+}
+
+@test "Simple foreach in if" {
+    cat << EOF | save_asset foreach_in_if.data
+{
+  "xs": [1,2,3,4,5,6,7,8,9],
+  "one": 1,
+  "limit": 6
+}
+EOF
+
+    cat << EOF | zexe foreach_in_if.zen foreach_in_if.data
+Given I have a 'float array' named 'xs'
+Given I have a 'float' named 'one'
+Given I have a 'float' named 'limit'
+When I create the new array
+If number 'one' is less than 'limit'
+Foreach 'x' in 'xs'
+When I insert 'x' in 'new array'
+EndForeach
+Endif
+Then print 'new array' as 'string'
+EOF
+    save_output "foreach_in_if.out"
+    assert_output '{"new_array":[1,2,3,4,5,6,7,8,9]}'
+}
+
+
+@test "Simple if in foreach" {
+    cat << EOF | save_asset if_in_foreach.data
+{
+  "xs": [1,2,3,4,5,6,7,8,9],
+  "limit": 6
+}
+EOF
+
+    cat << EOF | zexe if_in_foreach.zen if_in_foreach.data
+Given I have a 'float array' named 'xs'
+Given I have a 'float' named 'limit'
+When I create the new array
+Foreach 'x' in 'xs'
+If number 'x' is less than 'limit'
+When I insert 'x' in 'new array'
+Endif
+EndForeach
+Then print 'new array' as 'string'
+EOF
+    save_output "if_in_foreach.out"
+    assert_output '{"new_array":[1,2,3,4,5]}'
+}
+
+@test "Foreach with floats" {
+    cat << EOF | save_asset foreach_float.data
+{
+  "zero": 0,
+  "one": 2,
+  "limit": 11
+}
+EOF
+
+    cat << EOF | zexe foreach_float.zen foreach_float.data
+Given I have a 'float' named 'zero'
+Given I have a 'float' named 'one'
+Given I have a 'float' named 'limit'
+When I create the new array
+Foreach 'x' in sequence from 'zero' to 'limit' with step 'one'
+When I insert 'x' in 'new array'
+EndForeach
+Then print 'new array' as 'string'
+EOF
+    save_output "foreach_float.out"
+    assert_output '{"new_array":[0,2,4,6,8,10]}'
+}
+
+@test "Foreach with bigs" {
+    cat << EOF | save_asset foreach_big.data
+{
+  "zero": "-5",
+  "one": "2",
+  "limit": "11"
+}
+EOF
+
+    cat << EOF | zexe foreach_big.zen foreach_big.data
+Given I have a 'integer' named 'zero'
+Given I have a 'integer' named 'one'
+Given I have a 'integer' named 'limit'
+When I create the new array
+Foreach 'x' in sequence from 'zero' to 'limit' with step 'one'
+When I insert 'x' in 'new array'
+EndForeach
+Then print 'new array' as 'string'
+EOF
+    save_output "foreach_big.out"
+    assert_output '{"new_array":["-5","-3","-1","1","3","5","7","9","11"]}'
+}
+
```

Comment:
would be good to test multiple When statements inside foreach blocks with a bit more variety